### PR TITLE
Reference REST API for Zeebe where applicable

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

This PR adds references to the REST API in Zeebe where applicable, meaning primarily where we've migrated or introduced features using it. This means there are no changes yet for things like the Go client, workers, how to build your own client, etc. (admittedly that last one is omitted for time constraints :smile:)

One thing I noticed is that the examples are pretty outdated, and references files which are not in the repository anymore :upside_down_face: They're still useful honestly, though.

## When should this change go live?

Should be released with 8.5.0.

## PR Checklist

- [ ] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [x] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

**Comment by jessesimpson36** [comment]
I did a quick search to see if there were references to the old style of supplying `zeebe-gateway.ingress` but I didn't see any. I also looked at the changes and I'm happy with them.

**Comment by jessesimpson36** [comment]
oh my.  I was reading the wrong branch. I guess theres `np-rest-helm` and `np-rest-auth`. my mistake. please disregard my last message as it was intended for the other PR

**Review by christinaausley — APPROVED** [review]

File: docs/apis-tools/java-client/zeebe-process-test.md [hunk 1]
```diff
@@ -7,6 +7,10 @@ title: "Zeebe Process Test"
 processes. It will start a lightweight in-memory Zeebe engine and provide you with a set of assertions you can use to
 verify your process behaves as expected.
 
+:::note
+As of 8.5.0, Zeebe Process Test does not support the new REST API and related features (e.g. native user tasks).
```
Comment (tmetzke) [inline docs/apis-tools/java-client/zeebe-process-test.md:11]: 🔧 We should avoid non-inclusive terms like `native`.
```suggestion
As of 8.5.0, Zeebe Process Test does not support the new REST API and related features (e.g. Zeebe user tasks).
```

File: docs/apis-tools/zeebe-api-rest/zeebe-api-rest-authentication.md [hunk 2]
```diff
@@ -15,7 +15,7 @@ Your client must send a header in each request:
 For example, send a request using _curl_:
 
 ```shell
-curl -X POST -H -H :accept: application/json" -H "Authorization: Bearer <TOKEN>" -d '' http://localhost:8080/v1/topology
+curl -XGET -H'Accept: application/json' -H'Authorization: Bearer <TOKEN>' http://localhost:8080/v1/topology
```
Comment (tmetzke) [inline docs/apis-tools/zeebe-api-rest/zeebe-api-rest-authentication.md:18]: 🔧 I think we're missing a whitespace here, don't we?
```suggestion
curl -X GET -H'Accept: application/json' -H'Authorization: Bearer <TOKEN>' http://localhost:8080/v1/topology
```

File: docs/components/zeebe/technical-concepts/protocols.md [hunk 3]
```diff
@@ -35,3 +41,7 @@ If there is no client in your target language yet, you can [build your own clien
 
 Zeebe supports [loading arbitrary gRPC server interceptors](self-managed/zeebe-deployment/zeebe-gateway/interceptors.md) to intercept incoming
 calls.
+
+:::note
+As of 8.5.0, there is no interceptor or middleware functionality for the REST part of the client yet.
```
Comment (tmetzke) [inline docs/components/zeebe/technical-concepts/protocols.md:46]: ❓ Should we remove the indication that we'll actually add this? Underpromise, overdeliver 😄 
```suggestion
As of 8.5.0, there is no interceptor or middleware functionality for the REST part of the client.
```

**Review by tmetzke — DISMISSED** [review]
👍 The provided changes look good, nice adjustments.
🔧 I added some small suggestions, nothing major.
🤷 I can't really judge if we should adjust other places of the docs as well, time (and complaining users, I guess) will tell 😁

File: docs/apis-tools/zeebe-api-rest/zeebe-api-rest-authentication.md (hunk 2 above)
Comment (npepinpe) [reply docs/apis-tools/zeebe-api-rest/zeebe-api-rest-authentication.md:18]: AFAIK it doesn't matter. Same with the `-H'Authorization'` for example :shrug:

**Comment by npepinpe** [comment]
Need a new approval to merge after the changes :)

**Review by tmetzke — APPROVED** [review]

**Pull request merged**